docker.go

```diff
@@ -667,17 +667,22 @@ func (c *DockerContainer) StartLogProducer(ctx context.Context) error {
 				return
 			default:
 				h := make([]byte, 8)
-				_, err := r.Read(h)
+				_, err := io.ReadFull(r, h)
 				if err != nil {
 					// proper type matching requires https://go-review.googlesource.com/c/go/+/250357/ (go 1.16)
 					if strings.Contains(err.Error(), "use of closed network connection") {
 						now := time.Now()
 						since = fmt.Sprintf("%d.%09d", now.Unix(), int64(now.Nanosecond()))
 						goto BEGIN
 					}
-					// this explicitly ignores errors
-					// because we want to keep procesing even if one of our reads fails
-					continue
+					if errors.Is(err, context.DeadlineExceeded) {
+						// Probably safe to continue here
+						continue
+					}
+					_, _ = fmt.Fprintf(os.Stderr, "container log error: %+v", err)
+					// if we would continue here, the next header-read will result into random data...
+					<-stop
```

Comment:
`stop` is not valid anymore in this goroutine scope, as #947 refactored to call it without passing the `c.stopProducer`. Besides, the `c.stopProducer`'s type was changed to `context.CancelFunc`.

Reply:
I removed the stop writing right away in a hurry as I am in a workshop today. It's _not_ the ultimate solution as `r` stays open. Idea: `defer r.Close()`?